### PR TITLE
Feedback2

### PR DESCRIPTION
**Pull request by t3chguy**

### CODE REVIEW

File: src/github.com/t3chguy/riot-static/utils.go [hunk 1]
```diff
@@ -13,3 +13,16 @@
 // limitations under the License.
 
 package main
+
+import "strconv"
+
+func calcPaginationPage(pageString string, size int) (page int, skip int, end int) {
```
Comment (Kegsay) [inline src/github.com/t3chguy/riot-static/utils.go]: What is this supposed to do? Some docs would be nice. Is `size` the number of elements per page? The number of pages? What's `skip` and `end`?

File: src/github.com/t3chguy/riot-static/data.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	return *roomPointer, true
+}
+
+func (data *DataStore) SetRoom(roomID string, room *Room) {
+	data.Lock()
+	defer data.Unlock()
+	data.roomMap[roomID] = room
+}
+
+func min(a int, b int) int {
```
Comment (Kegsay) [inline src/github.com/t3chguy/riot-static/data.go]: Why is this defined here? You have a utils file. You use this in this file as well as in `src/github.com/t3chguy/riot-static/rooms.go`.

File: src/github.com/t3chguy/riot-static/data.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	data.RUnlock()
+
+	if roomPointer == nil {
+		return Room{}, false
+	}
+	exists = true
+
+	data.RLock()
+	expired := roomPointer.Cached.CheckExpired()
+	data.RUnlock()
```
Comment (Kegsay) [inline src/github.com/t3chguy/riot-static/data.go]: Right, I was going to let you off on this, but this has convinced me not to. You need to tell the reader what your mutex is supposed to be protecting (a comment or an obvious name e.g. `roomMapMutex`).

Why the fuss you may ask? Well, without it, I cannot verify if you're actually protecting what you think you are protecting. I thought the embedded mutex was protecting the `roomMap`, but then why are you taking a lock out here when you don't access it?

Locks are hard enough to get right as it is, don't make it harder for yourself or me.

File: src/github.com/t3chguy/riot-static/data.go (hunk 2 above)
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/data.go]: Fixed

File: src/github.com/t3chguy/riot-static/data.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func (data *DataStore) GetRoomList(start int, end int) []*Room {
+	data.RLock()
+	defer data.RUnlock()
+	length := data.GetNumRooms()
```
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/data.go]: Fixed

File: src/github.com/t3chguy/riot-static/mxc.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+// mxcRegex allows splitting an mxc into a serverName and mediaId
+// FindStringSubmatch of which results in [_, serverName, mediaId] if valid
+// and [] if invalid mxc is provided.
+// Examples:
+// "mxc://foo/bar" => ["mxc://foo/bar", "foo", "bar"]
+// "mxc://bar/foo#auto" => ["mxc://bar/foo#auto", "bar", "foo"]
+// "invalidMxc://whatever" => [] (Invalid MXC Caught)
+var mxcRegex = regexp.MustCompile(`mxc://(.+?)/(.+?)(?:#.+)?$`)
+
+type MxcUrl string
```
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/mxc.go]: Fixed

File: src/github.com/t3chguy/riot-static/riot-static.go [hunk 6]
```diff
@@ -12,223 +12,212 @@
...
 
-	data.RLock()
-	c.HTML(http.StatusOK, "room.html", gin.H{
-		"Room": data.Rooms[roomId],
-	})
-	data.RUnlock()
-}
+	// filter on actually WorldReadable publicRooms
+	for _, x := range resp.Chunk {
+		if x.WorldReadable {
```
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/riot-static.go]: Fixed

File: src/github.com/t3chguy/riot-static/data.go (hunk 3 above)
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/data.go]: Locking `DataStore` ensures that it and all the Rooms (originals) it contains are locked, you mentioned `SetRoom` could get around this, it was from my first draft of the `DataStore` and wasn't a used method.

File: src/github.com/t3chguy/riot-static/client.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+			}
+			if err = json.Unmarshal(data, &powerLevels); err != nil {
+				return
+			}
+		}
+	}
+	return
+}
+
+func /*(cli *client)*/ peekRoom(roomId string) (initialSync *RespInitialSync, memberInfo map[string]*MemberInfo, powerLevels *PowerLevelsEvent, err error) {
```
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/client.go]: Fixed

File: src/github.com/t3chguy/riot-static/riot-static.go [hunk 8]
```diff
@@ -12,223 +12,212 @@
...
+		Handler:      router,
+		Addr:         ":" + port,
+	}
+
+	panic(srv.ListenAndServe())
+}
+
+const LoadPublicRoomsPeriod = time.Hour
+
+func runCron() {
```
Comment (t3chguy) [inline src/github.com/t3chguy/riot-static/riot-static.go]: Fixed

**Pull request closed without merging**